api/task_manager_test.cc

```diff
@@ -0,0 +1,107 @@
+        co_await ctx.tm.invoke_on_all([name = "test"] (task_manager& tm) -> future<> {
+            auto& modules = tm.get_modules();
+            auto it = modules.find(name);
+            if (it != modules.end()) {
+                co_await it->second->stop();
+            }
+        });
+        co_return json_void();
+    });
+
+    tmt::register_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        sharded<task_manager>& tms = ctx.tm;
+        auto it = req->query_parameters.find("task_id");
+        task_id id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_random_id();
+        it = req->query_parameters.find("keyspace");
+        std::string keyspace = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("table");
+        std::string table = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("type");
+        std::string type = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("entity");
+        std::string entity = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("sequence_number");
+        int sequence_number = it != req->query_parameters.end() ? boost::lexical_cast<int>(it->second) : 0;
+        it = req->query_parameters.find("parent_id");
+        task_id parent_id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_null_id();
+        co_await tms.invoke_on(0, [id, name = "test", sequence_number, keyspace = std::move(keyspace), table = std::move(table), type = std::move(type), entity = std::move(entity), parent_id] (task_manager& tm) -> future<> {
+            auto& modules = tm.get_modules();
+            auto it = modules.find(name);
+            auto _id = id;
+            if (it != modules.end()) {
+                // TODO Deexie use make_task
+                auto task_impl_ptr = std::make_unique<test_task>(it->second, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity));
+                auto task = seastar::make_lw_shared<task_manager::task>(std::move(task_impl_ptr));
+                task->start();
+            } else {
+                co_await seastar::coroutine::return_exception(std::runtime_error("module not found"));
+            }
+        });
+        co_return id.to_sstring();
+    });
+
+    tmt::unregister_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        task_id id = task_id{utils::UUID{req->query_parameters["task_id"]}};
+        sharded<task_manager>& tms = ctx.tm;
+        task_manager::foreign_task_ptr task = co_await task_manager::lookup_task_on_all_shards(tms, id);
+        co_await tms.invoke_on(task.get_owner_shard(), [name = "test", t = std::move(task), id] (auto& tm) mutable {
```

Comment:
name seems to be unused?
(and also can be defined as a regular statement inside the lambda function)